### PR TITLE
Add search modifiers in Search | Fix #7967

### LINKED ISSUE #7967: Search for text with braces in pattern.
When i try to search in source (ctrl+shift+f) wth braces in pattern i find nothing 
ex: `parameters = Object.assign({node: null}, parameters);`
Then i must escape all braces, which is very annoying, to find what i need.
ex: `parameters = Object.assign\({node: null}, parameters\);`

I propose to search with text pattern by default (like in old versions). Regex search pattern should be an option.

### PR DESCRIPTION
**Pull request by charanjit-singh**
Fixes #7967 

### Summary of Changes

* Added Regex, Case Sensitive, Whole word modifiers in search ( Ctrl + shift + F )
* Added styles for active classes on modifier buttons.

### Test Plan

Tests can be done by searching using combination of all modifiers and checking their results.

Example test plan:

- [x] Open test file in browser.
- [x] Press Ctrl + Shift + F for searching.
- [x] In Input box, type the words to test.
- [x] Check/ uncheck the modifiers.
- [x] Verify the expected results.

### Screenshots
screenshot from 2019-02-26 17-24-15
---

screenshot from 2019-02-26 17-24-23

---

screenshot from 2019-02-26 17-24-44

### CODE REVIEW

**Comment by charanjit-singh** [comment]
Tests are failing as we need to update tests also.

**Comment by darkwing** [comment]
Thank you for submitting @charanjit-singh!  Quick note:  Jest tests are failing because of old snapshots; you can run `yarn jest -u` to update those and add as a second commit! 👍

File: src/utils/sources-tree/sortTree.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ export function sortEntireTree(
   tree: TreeDirectory,
   debuggeeUrl: string = ""
 ): TreeDirectory {
+  console.log("Sorting ENtire Tree;");
```
Comment (darkwing) [inline src/utils/sources-tree/sortTree.js]: Ooops, I don't think we want this. :)

**Comment by darkwing** [comment]
Functionally this works great!  One quick style nit:

When a button is active (or turned on) for find in file, the icon is blue and shows a light hover color; in your PR, the icon doesn't change color and the highlighted color is darker.  Let's do the same styling as the find in file.

This is really impressive @charanjit-singh !  Great work!

**Comment by charanjit-singh** [comment]
Okay I'll update it soon.
Thanks for reviewing.

File: src/utils/sources-tree/sortTree.js (hunk 1 above)
Comment (charanjit-singh) [reply src/utils/sources-tree/sortTree.js]: Oh sorry

**Comment by charanjit-singh** [comment]
@darkwing  I'm unable to update tests using `yarn jest -u`.  Help !

**Comment by darkwing** [comment]
Hello @charanjit-singh!  I can update jest tests but I need you to check out a few of the test failures -- it looks like results may be being lost

**Comment by charanjit-singh** [comment]
Hi !, 
There was a problem. 
Existing `ProjectSearch` was not in Redux, That's why, The way, I was implementing was failing tests.
So, I shifted back to `master`, added `redux` and fixed above issue.
 Now I'm  creating a new pull request from new branch. Kindly Delete this Pull request.

**Comment by charanjit-singh** [comment]
Pull Request #8051

**Pull request closed without merging**